### PR TITLE
docs: explain that in-flight transaction inputs/outputs are not available to spent on child chain

### PR DESCRIPTION
**Pull request by pgebal**
Closes #533
Changes:
 * rename `ExitGameController.isOutputSpent` to `ExitGameController.isOutputFinalized` to avoid confusion
 * add section in integration docs explaining when in-flight transaction inputs/ouputs are not available to spend on child chain

### CODE REVIEW

File: plasma_framework/contracts/src/framework/ExitGameController.sol [hunk 1]
```diff
@@ -187,9 +187,9 @@ contract ExitGameController is ExitGameRegistry {
      * @notice Checks whether any of the output with the given outputIds is already spent
      * @param _outputIds Output IDs to check
      */
-    function isAnyOutputsSpent(bytes32[] calldata _outputIds) external view returns (bool) {
+    function isAnyOutputsFinalized(bytes32[] calldata _outputIds) external view returns (bool) {
```
Comment (kevsul) [inline plasma_framework/contracts/src/framework/ExitGameController.sol]: Strictly speaking this should be `isAnyOutputFinalized`.
But it's not a big deal, I'll leave it up to you...

File: plasma_framework/docs/integration-docs/integration-doc.md [hunk 2]
```diff
@@ -629,6 +629,10 @@ PaymentExitGame.startInFlightExit([
 
 ### Piggybacking on an in-flight exit input / output
 
+To exit input or output of an in-flight transaction user has to piggyback on it.
```
Comment (kevsul) [inline plasma_framework/docs/integration-docs/integration-doc.md]:
```suggestion
To exit an input or output of an in-flight transaction the user has to piggyback on it.
```

File: plasma_framework/docs/integration-docs/integration-doc.md [hunk 3]
```diff
@@ -629,6 +629,10 @@ PaymentExitGame.startInFlightExit([
...
+
+When in-flight transaction exits, child chain removes the inputs of the in-flight transaction from the spendable set and it no longer can be spent. Calls to child chain to spend the inputs or outputs of the in-flight transaction result in `submit:utxo_not_found` error. The effect of input or output not being available to spend is not visible in contract's state until exit is processed. After the in-flight exit is processed `PlasmaFramework.isOutputFinalized(piggybackedInputOrOutputPosition)` is set to false.
```
Comment (kevsul) [inline plasma_framework/docs/integration-docs/integration-doc.md]:
```suggestion
When in-flight transaction exits, the child chain removes the inputs of the in-flight transaction from the spendable set and they can no longer be spent. Calls to the child chain to spend the inputs or outputs of an in-flight transaction result in `submit:utxo_not_found` error. The effect of an input or output not being available to spend is not visible in contract's state until the exit is processed. After the in-flight exit is processed `PlasmaFramework.isOutputFinalized(piggybackedInputOrOutputPosition)` is set to true.
```

Comment (pgebal) [reply plasma_framework/docs/integration-docs/integration-doc.md]: What a mistake I've made with this `is set to false`!

**Pull request merged**